lang/c/examples/quickstop.c

```diff
@@ -25,214 +25,69 @@
-
-	avro_value_t person;
-	avro_generic_value_new(person_class, &person);
-
-	int rval;
-
-	rval = avro_file_reader_read_value(db, &person);
-	if (rval == 0) {
-		int64_t id;
-		int32_t age;
-		int32_t *p;
-		size_t size;
-		avro_value_t id_value;
-		avro_value_t first_value;
-		avro_value_t last_value;
-		avro_value_t age_value;
-		avro_value_t phone_value;
-
-		if (avro_value_get_by_name(&person, "ID", &id_value, NULL) == 0) {
-			avro_value_get_long(&id_value, &id);
-			fprintf(stdout, "%"PRId64" | ", id);
-		}
-		if (avro_value_get_by_name(&person, "First", &first_value, NULL) == 0) {
-			avro_value_get_string(&first_value, &p, &size);
-			fprintf(stdout, "%15s | ", p);
-		}
-		if (avro_value_get_by_name(&person, "Last", &last_value, NULL) == 0) {
-			avro_value_get_string(&last_value, &p, &size);
-			fprintf(stdout, "%15s | ", p);
-		}
-		if (avro_value_get_by_name(&person, "Phone", &phone_value, NULL) == 0) {
-			avro_value_get_string(&phone_value, &p, &size);
-			fprintf(stdout, "%15s | ", p);
-		}
-		if (avro_value_get_by_name(&person, "Age", &age_value, NULL) == 0) {
-			avro_value_get_int(&age_value, &age);
-			fprintf(stdout, "%"PRId32" | ", age);
-		}
-		fprintf(stdout, "\n");
-
-		/* We no longer need this memory */
-		avro_value_decref(&person);
-		avro_value_iface_decref(person_class);
-	}
-	return rval;
-}
-
-int main(void)
-{
-	int rval;
-	avro_file_reader_t dbreader;
-	avro_file_writer_t db;
-	avro_schema_t projection_schema, first_name_schema, phone_schema;
-	int64_t i;
-	const char *dbname = "quickstop.db";
-	char number[15] = {0};
-
-	/* Initialize the schema structure from JSON */
-	init_schema();
-
-	/* Delete the database if it exists */
-	remove(dbname);
-	/* Create a new database */
-	rval = avro_file_writer_create_with_codec
-	    (dbname, person_schema, &db, QUICKSTOP_CODEC, 0);
-	if (rval) {
-		fprintf(stderr, "There was an error creating %s\n", dbname);
-		fprintf(stderr, " error message: %s\n", avro_strerror());
-		exit(EXIT_FAILURE);
-	}
-
-	/* Add lots of people to the database */
-	for (i = 0; i < 1000; i++)
+	const char *avro_file = "f_sgeb25868e7855fe7a1d7f446f07923744_14d_v1609459200000_ts1609474609422.avro";
+	avro_file_buffer_reader_t db;
+	if (avro_file_buffer_reader(avro_file, &db))
 	{
-		sprintf(number, "(%d)", (int)i);
-		add_person(db, "Dante", "Hicks", number, 32);
-		add_person(db, "Randal", "Graves", "(555) 123-5678", 30);
-		add_person(db, "Veronica", "Loughran", "(555) 123-0987", 28);
-		add_person(db, "Caitlin", "Bree", "(555) 123-2323", 27);
-		add_person(db, "Bob", "Silent", "(555) 123-6422", 29);
-		add_person(db, "Jay", "???", number, 26);
+		fprintf(stdout, "Error opening file: %s\n", avro_strerror());
+		exit(EXIT_FAILURE);
 	}
 
-	/* Close the block and open a new one */
-	avro_file_writer_flush(db);
-	add_person(db, "Super", "Man", "123456", 31);
-
-	avro_file_writer_close(db);
+	avro_datum_t* mapping_datumn;
+	char* json;
+	create_fields_mapping_block_schema(db, &mapping_datumn);
```

Comment:
The mapping datum looks like:
`{"__offset": 5268, "showings": [5327, 78215, 145395, 212067, 289138, 355771, 420683, 484106, 555186, 626202, 694909, 758265, 822154, 886024, 949542, 1013952, 1078836, 1142333, 1206578, 1268554, 1332904, 1393106, 1453917, 1514625, 1578569, 1643499, 1706092, 1770526, 1836827, 1903127, 1965962, 2030854, 2104378, 2172404, 2238150, 2304381, 2363956, 2427219, 2490676, 2552573, 2618703, 2683670, 2750019, 2818217, 2880199, 2942180, 3008144, 3072108, 3142399, 3214448, 3283377, 3353277, 3418246, 3479095, 3539957, 3604199, 3667041, 3730212, 3795375, 3864379, 3930267, 3993553, 4059051, 4124860, 4186521, 4253472, 4322411, 4388060, 4453839, 4519555, 4584551, 4646041, 4708509, 4770862, 4833814, 4895074, 4961393, 5029539, 5093668, 5162270, 5228429, 5294368, 5357680, 5421804, 5493349, 5558889, 5622914, 5689388, 5760819, 5825671, 5886031, 5946164, 6011946, 6077901, 6147696, 6210476, 6275992, 6340979, 6411680, 6471832, 6531971, 6592084, 6652288, 6715388, 6778058, 6842468, 6907025, 6968381, 7034161, 7097108, 7170851, 7254704, 7322481, 7382962, 7447158, 7513595, 7583046, 7647043, 7709689, 7769893, 7835234, 7901709, 7968959, 8031656, 8107763, 8174528, 8239641, 8305535, 8371159, 8430989, 8491195, 8554961, 8625068, 8693157, 8763930, 8826760, 8887954, 8947354, 9012314, 9085160, 9150498, 9227106, 9302686, 9365032, 9430218, 9493585, 9557205, 9619345, 9680084, 9740537, 9802666, 9867954, 9932149, 9994222, 10054571, 10117468, 10186982, 10247902, 10311255, 10378630, 10439316, 10501578, 10568470, 10632098, 10703192, 10782825, 10850513, 10926494, 10994828, 11064267, 11124665, 11188438, 11252563, 11313033, 11385102, 11449536, 11512948, 11572794, 11635174, 11702635, 11767074, 11830132, 11895468, 11961550, 12027703, 12091411, 12151664, 12213601, 12278794, 12344698, 12409838, 12475399, 12541221, 12615237, 12681068, 12744935, 12807088, 12868936, 12946991, 13008552, 13074293, 13140909, 13208486, 13275919, 13348584, 13423005, 13484147, 13552863, 13622180, 13688255, 13750625, 13815510, 13878299, 13941398, 14006515, 14068786, 14130196, 14190718, 14253944, 14317429, 14378926, 14440896, 14503245, 14564651, 14633324, 14693201, 14752148, 14811499, 14875319, 14940545, 15023497, 15096994, 15179718, 15248143, 15311308, 15394966, 15455670, 15523109, 15598286, 15669818, 15744823, 15814288, 15884411, 15949126, 16014643, 16079832, 16155191, 16221246, 16297446, 16366392, 16428075, 16504130, 16586375, 16668480, 16743307, 16825724, 16901520, 16988934, 17073074, 17143979, 17206228, 17271542, 17337025, 17416155, 17486527, 17551043, 17628156, 17700659, 17766570, 17834819, 17917546, 18010912, 18094219, 18171088, 18235375, 18298864, 18365769, 18429315, 18498836, 18567443, 18641765, 18709494, 18776604, 18854173, 18932087, 19008693, 19072869, 19146199, 19209569, 19284674, 19357744, 19431171, 19508337, 19586380, 19663796, 19741514, 19818965, 19884124, 19949291, 20021154, 20089146, 20161454, 20226876, 20303621, 20366869, 20428957, 20500231, 20561149, 20623473, 20686459, 20751717, 20815037, 20881959, 20956258, 21021477, 21092625, 21170698, 21237086, 21311420, 21390545, 21469185, 21543116, 21605643, 21667501, 21729829, 21799215, 21864324, 21932494, 22018356, 22086730, 22152759, 22224908, 22286302, 22347444, 22420817, 22484965, 22548762, 22612240, 22675274, 22739952, 22800709, 22865303, 22933650, 23002211, 23068220, 23131608, 23192102, 23252570, 23313190, 23373813, 23441350, 23502815, 23572650, 23650920, 23719722, 23796592, 23859740, 23925141, 24000572, 24071969, 24143181, 24206753, 24270516, 24333376, 24394050, 24461347, 24528511, 24602735, 24665431, 24726679, 24788069, 24853178, 24918098, 24979594, 25041364, 25105499, 25174916, 25239538, 25301556, 25373757, 25451093, 25516771, 25596283, 25663669, 25726950, 25801471, 25875642, 25952506, 26023989, 26101585, 26178343, 26245510, 26307256, 26370308, 26432252, 26495310, 26559427, 26621990, 26692799, 26762507, 26833418, 26899456, 26966414, 27031799, 27105167, 27171418, 27246527, 27319221, 27395698, 27470160, 27539918, 27610307, 27669778, 27729923, 27790161, 27853407, 27915753, 27981400, 28045828, 28107338, 28166704, 28229646, 28290526, 28353885, 28414605, 28477470, 28543690, 28606258, 28672187, 28734957, 28795840, 28855588, 28915279, 28975969, 29038721, 29099651, 29159963, 29220319, 29289418, 29357004, 29426367, 29426623], "contents": [29426627, 29729065, 29994892, 30244534, 30515798, 30799530, 31081711, 31358107, 31630685, 31894184, 32181693, 32478629, 32739988, 33030264, 33339609, 33641403, 33936575, 34205276, 34491928, 34785632, 35058656, 35339540, 35618810, 35909476, 36201745, 36496635, 36820108, 37168388, 37486753, 37768004, 38081216, 38355135, 38640920, 38884351, 39193157, 39473904, 39731050, 40012555, 40296343, 40613123, 40890039, 41182299, 41463563, 41766611, 42066567, 42339783, 42584869, 42850439, 43139596, 43442628, 43786887, 44083935, 44375291, 44656793, 44975321, 45294554, 45602295, 45893669, 46196568, 46476227, 46747181, 47023761, 47303541, 47575119, 47871015, 48120476, 48387364, 48667100, 48962081, 49219760, 49489492, 49759595, 50034241, 50344176, 50640194, 50926111, 51200930, 51483030, 51731158, 52012911, 52315389, 52593916, 52858110, 53156566, 53476484, 53743265, 54019531, 54306574, 54608091, 54895618, 55199768, 55543041, 55860423, 56154736, 56444155, 56779382, 57089919, 57358418, 57643653, 57781227], "collections": [57781231, 57857590, 57933879, 58009101, 58084878, 58160474, 58237118, 58311649, 58387101, 58463812, 58539069, 58614888, 58690295, 58765014, 58840577, 58915476, 58991204, 59068263, 59143520, 59219073, 59294190, 59368756, 59443299, 59518618, 59594567, 59670081, 59746297, 59821622, 59896780, 59972152, 60046540, 60120950, 60196004, 60225795], "showingDeletes": [], "contentDeletes": [], "collectionDeletes": []}
`

Reply:
It contains the begin-end offsets in bytes from the file begin for each chunk. For example: offsets  [begin: 29426627, end: 2972906] should be used to get the first chunk of the "contents" array [0-512], - [begin: 29729065, end: 29994892]  to get the second one [512-1024]